android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimeSpinnerNumberPicker.java

```diff
@@ -0,0 +1,234 @@
+        hoursWheel.setDescendantFocusability(NumberPicker.FOCUS_BLOCK_DESCENDANTS);
+        hoursWheel.setDisplayedValues(hoursString);
+        hoursWheel.setMaxValue(hoursString.length - 1);
+        hoursWheel.setMinValue(0);
+        minutesString = generateNumbers(0, 59, formatter, 6, minuteInterval);
+        minutesWheel.setDescendantFocusability(NumberPicker.FOCUS_BLOCK_DESCENDANTS);
+        minutesWheel.setDisplayedValues(minutesString);
+        minutesWheel.setMaxValue(minutesString.length - 1);
+        minutesWheel.setMinValue(0);
+        hoursWheel.setOnValueChangedListener(this);
+        minutesWheel.setOnValueChangedListener(this);
+        amPmWheel = null;
+
+        if ( !format24 ) {
+            amPmWheel = makeAmPmWheel(activity);
+        }
+
+        LinearLayout layout = new LinearLayout(activity);
+        layout.setOrientation(LinearLayout.HORIZONTAL);
+        layout.addView(hoursWheel);
+        layout.addView(minutesWheel);
+        if ( !format24 ) {
+            layout.addView(amPmWheel);
+        }
+
+        setNativeView(layout);
+    }
+
+    private String[] generateNumbers(int minValue, int maxValue, NumberFormat formatter, int maxCharLength, int stepValue) {
+        int itemCount = ( (maxValue - minValue) / stepValue) + 1;
+        List<String> list = new ArrayList<String>();
+        for (int index = 0 ; index < itemCount; index++) {
+            int actualValue = minValue + index * stepValue;
+            if (formatter != null) {
+                list.add(formatter.format(actualValue));
+            } else {
+                list.add(Integer.toString(actualValue));
+            }
+        }
+        return list.toArray(new String[0]);
+    }
+
+    @Override
+    public void processProperties(KrollDict d) {
+        super.processProperties(d);
+
+        boolean valueExistsInProxy = false;
+
+        if (d.containsKey(TiC.PROPERTY_VALUE)) {
+            calendar.setTime((Date)d.get(TiC.PROPERTY_VALUE));
+            valueExistsInProxy = true;
+        }
+
+        setValue(calendar.getTimeInMillis() , true);
+
+        if (!valueExistsInProxy) {
+            proxy.setProperty(TiC.PROPERTY_VALUE, calendar.getTime());
+        }
+
+    }
+
+    @Override
+    public void propertyChanged(String key, Object oldValue, Object newValue, KrollProxy proxy) {
+        if (key.equals(TiC.PROPERTY_VALUE)) {
+            Date date = (Date)newValue;
+            setValue(date.getTime());
+        }
+
+        super.propertyChanged(key, oldValue, newValue, proxy);
+    }
+
+    public void setValue(long value) {
+        setValue(value, false);
+    }
+
+    public void setValue(long value, boolean suppressEvent) {
+        boolean format24 = true;
+        if (proxy.hasProperty("format24")) {
+            format24 = TiConvert.toBoolean(proxy.getProperty("format24"));
+        }
+        calendar.setTimeInMillis(value);
+
+        suppressChangeEvent = true;
+        ignoreItemSelection = true;
```

Comment:
It looks like you're running this on KrollRuntime thread, triggering the callback on Main thread. Wouldn't it be better to run this function on main thread so we don't have to set the same variable back and forth mulitple times?